File: src/main/java/org/theoliverlear/entity/user/personal/Interest.java
package org.theoliverlear.entity.user.personal;
//=================================-Imports-==================================
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Table(name = "interests")
@Entity
public class Interest {
    //============================-Variables-=================================
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "interest_title")
    private String interestTitle;
    @ManyToOne
    @JoinColumn(name = "interests_id")
    private Interests interests;
    //============================-Constants-=================================
    @Transient
    @JsonIgnore
    public static final Interest SOFTWARE_DEVELOPMENT = new Interest("Software Development");
    @Transient
    @JsonIgnore
    public static final Interest WEB_DEVELOPMENT = new Interest("Web Development");
    @Transient
    @JsonIgnore
    public static final Interest MOBILE_DEVELOPMENT = new Interest("Mobile Development");
    @Transient
    @JsonIgnore
    public static final Interest DATABASE_MANAGEMENT = new Interest("Database Management");
    @Transient
    @JsonIgnore
    public static final Interest CYBERSECURITY = new Interest("Cybersecurity");
    @Transient
    @JsonIgnore
    public static final Interest DATA_SCIENCE = new Interest("Data Science");
    @Transient
    @JsonIgnore
    public static final Interest GAME_DEVELOPMENT = new Interest("Game Development");
    @Transient
    @JsonIgnore
    public static final Interest TEAM_BUILDING = new Interest("Team Building");
    @Transient
    @JsonIgnore
    public static final Interest PROJECT_MANAGEMENT = new Interest("Project Management");
    @Transient
    @JsonIgnore
    public static final Interest ARTIFICIAL_INTELLIGENCE = new Interest("Artificial Intelligence");
    //===========================-Constructors-===============================
    public Interest() {
        this.interestTitle = null;
    }
    public Interest(String interestTitle) {
        this.interestTitle = interestTitle;
    }
    public Interest(String interestTitle, Interests interests) {
        this.interestTitle = interestTitle;
        this.interests = interests;
    }
    //============================-Overrides-=================================

    //------------------------------Equals------------------------------------
    @Override
    public boolean equals(Object object) {
        if (object == this) {
            return true;
        }
        if (object instanceof Interest interest) {
            return this.interestTitle.equals(interest.interestTitle);
        }
        return false;
    }
    //------------------------------Hash-Code---------------------------------
    @Override
    public int hashCode() {
        return this.interestTitle.hashCode();
    }
}

File: src/main/resources/static/script/Interest.ts
export enum Interest {
    SOFTWARE_DEVELOPMENT = 'Software Development',
    WEB_DEVELOPMENT = 'Web Development',
    MOBILE_DEVELOPMENT = 'Mobile Development',
    DATABASE_MANAGEMENT = 'Database Management',
    CYBERSECURITY = 'Cybersecurity',
    DATA_SCIENCE = 'Data Science',
    GAME_DEVELOPMENT = 'Game Development',
    TEAM_BUILDING = 'Team Building',
    PROJECT_MANAGEMENT = 'Project Management',
    ARTIFICIAL_INTELLIGENCE = 'Artificial Intelligence'
}
export namespace Interest {
    export function from(interestString: string): Interest {
        let interest: Interest;
        switch(interestString) {
            case Interest.SOFTWARE_DEVELOPMENT:
                interest = Interest.SOFTWARE_DEVELOPMENT;
                break;
            case Interest.WEB_DEVELOPMENT:
                interest = Interest.WEB_DEVELOPMENT;
                break;
            case Interest.MOBILE_DEVELOPMENT:
                interest = Interest.MOBILE_DEVELOPMENT;
                break;
            case Interest.DATABASE_MANAGEMENT:
                interest = Interest.DATABASE_MANAGEMENT;
                break;
            case Interest.CYBERSECURITY:
                interest = Interest.CYBERSECURITY;
                break;
            case Interest.DATA_SCIENCE:
                interest = Interest.DATA_SCIENCE;
                break;
            case Interest.GAME_DEVELOPMENT:
                interest = Interest.GAME_DEVELOPMENT;
                break;
            case Interest.TEAM_BUILDING:
                interest = Interest.TEAM_BUILDING;
                break;
            case Interest.PROJECT_MANAGEMENT:
                interest = Interest.PROJECT_MANAGEMENT;
                break;
            case Interest.ARTIFICIAL_INTELLIGENCE:
                interest = Interest.ARTIFICIAL_INTELLIGENCE;
                break;
            default:
                throw new Error('Error in parsing Interest');
        }
        return interest;
    }
}